### PR TITLE
[master][WIP] Duplicate `arm64` repo paths for rpm-based distros to `aarch64` paths 

### LINKED ISSUE #65198: [TECH DEBT] RPM based distros should use aarch64 instead of arm64 on repo.saltproject.io
### Description of the tech debt to be addressed, include links and screenshots
Currently, we have paths for rpm based distros that look like this: https://repo.saltproject.io/salt/py3/photon/4/arm64/.
For each rpm based distro, we should replace `arm64` with `aarch64`. The old style paths will be deprecated properly, and phase EOL when 3009.0 is released.

Tasklist:

- [ ] Update repo publishing to publish to both paths
- [ ] Update the package tests to prefer `aarch64` paths
- [ ] Update the download tests to _only_ pull from `aarch64` paths
- [ ] Deprecation notice in the install guide, and have it prefer `aarch64` paths

### PR DESCRIPTION
**Pull request by MKLeb**
### What does this PR do?
See title

### What issues does this PR fix or reference?
Fixes: https://github.com/saltstack/salt/issues/65198

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by MKLeb** [comment]
Staging run here: https://github.com/MKLeb/salt/actions/runs/6617571464

Errors should be fixed by: https://github.com/saltstack/salt/pull/65374

File: .github/workflows/test-packages-action-macos.yml [hunk 1]
```diff
@@ -70,7 +70,10 @@ jobs:
 
   generate-matrix:
     name: Generate Matrix
-    runs-on: ${{ github.event.repository.private && fromJSON('["self-hosted", "linux", "x86_64"]') || 'ubuntu-latest' }}
+    runs-on:
+      - self-hosted
+      - linux
+      - x86_64
```
Comment (s0undt3ch) [inline .github/workflows/test-packages-action-macos.yml:76]: Why are we changing this part? I think we talked about it but remind me please...

File: tools/ci.py [hunk 2]
```diff
@@ -741,53 +726,108 @@ def pkg_matrix(
...
-                for version in testing_releases
-                if version < tools.utils.Version("3006.0")
-            ]
-        for version in versions:
-            _matrix.append(
-                {
-                    "test-chunk": session,
-                    "version": version,
-                }
+    adj_versions = []
```
Comment (s0undt3ch) [inline tools/ci.py:734]: What does `adj` stand for?

File: tools/ci.py [hunk 3]
```diff
@@ -741,53 +726,108 @@ def pkg_matrix(
...
+            ctx.info(f"No {version} ({backend}) for {distro_slug} at {prefix}")
+
     ctx.info("Generated matrix:")
-    ctx.print(_matrix, soft_wrap=True)
+    ctx.print(matrix, soft_wrap=True)
 
     if github_output is not None:
         with open(github_output, "a", encoding="utf-8") as wfh:
-            wfh.write(f"matrix={json.dumps(_matrix)}\n")
+            wfh.write(f"matrix={json.dumps(matrix)}\n")
```
Comment (s0undt3ch) [inline tools/ci.py:830]: Make this `_matrix` against, it obfuscates the `matrix` method...

File: .github/workflows/test-packages-action-macos.yml (hunk 1 above)
Comment (MKLeb) [reply .github/workflows/test-packages-action-macos.yml:76]: it needs to be able to access the s3 bucket, so it needs our runners for the creds

**Comment by MKLeb** [comment]
Closing in favor of https://github.com/saltstack/salt/pull/65553

**Pull request closed without merging**